var/spack/repos/builtin/packages/libcircle/package.py

```diff
@@ -22,6 +22,12 @@ class Libcircle(AutotoolsPackage):
     depends_on('pkgconfig', type='build')
     depends_on('libpciaccess', type='link')
 
+    patch('CrayPE_configure-ac.patch', when='%cce')
+
+    @when('%cce')
+    def autoreconf(self, spec, prefix):
```

Comment:
Have you considered using `force_autoreconf` as described with the second example at https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html?highlight=force_autoreconf#force-autoreconf?

Reply:
In other words, something along the lines:

```
    @property
    def force_autoreconf(self):
        return self.spec.satisfies('%cce')
```